sklearn/preprocessing/imputation.py

```diff
@@ -372,4 +387,15 @@ def transform(self, X):
 
             X[coordinates] = values
 
+            features_with_missing_values = np.where(np.any(mask, axis=0))[0]
+            imputed_mask = mask[:, features_with_missing_values]
+
+            if self.axis == 0:
+                self.imputed_features_ = valid_statistics_indexes[
```

Comment:
`valid_statistics_indexes` is poorly named. It has no statistics. Can we just call it `valid_idx`?